gammapy/datasets/simulate.py

```diff
@@ -26,6 +27,66 @@ class MapDatasetEventSampler:
     def __init__(self, random_state="random-seed"):
         self.random_state = get_random_state(random_state)
 
+    def _sample_coord_time_energy(self, dataset, temporal_model, t_delta="1 sec"):
```

Comment:
Do you need to pass the full dataset or simply the `MapEvaluator` as in the other `sample_coord_energy`?

Reply:
Ah right, we can just pass the `MapEvaluator`. I've changed it.